src/main/webapp/app/exam/exam-scores/exam-scores.component.ts

```diff
@@ -713,4 +777,19 @@ export class ExamScoresComponent implements OnInit, OnDestroy {
         console.log(errorMessage);
         Sentry.captureException(new Error(errorMessage));
     }
+
+    private static declareExerciseType(exerciseInfo: ExerciseInfo) {
+        switch (exerciseInfo.exerciseType) {
+            case 'TextExercise':
+                return ExerciseType.TEXT;
+            case 'ModelingExercise':
+                return ExerciseType.MODELING;
+            case 'ProgrammingExercise':
+                return ExerciseType.PROGRAMMING;
+            case 'FileUploadExercise':
+                return ExerciseType.FILE_UPLOAD;
+            case 'QuizExercise':
+                return ExerciseType.QUIZ;
+        }
+    }
```

Comment:
can't be initially set the type to an enum type in `exam-score-dtos.model.ts`? I am not sure about this, just a thought.

Reply:
We cannot get around mapping the actual Java Class name to the shorter name (quiz instead of QuizExercise) for the URL. Do you think mapping it server-side has an advantage?

Reply:
no just leave it as it is... thanks for clarification, I was a bit confused, I think the way you implemented it is reasonable.

Reply:
👍